src/main/webapp/app/exercises/programming/manage/update/update-components/custom-build-plans/programming-exercise-custom-aeolus-build-plan.component.ts

```diff
@@ -160,12 +142,41 @@ export class ProgrammingExerciseCustomBuildPlanComponent implements OnChanges {
         }
     }
 
+    addParameter(): void {
+        if (this.active) {
+            if (!this.active.parameters) {
+                this.active.parameters = new Map<string, string | boolean | number>();
+            }
+            this.active.parameters.set('newParameter' + this.active.parameters.size, 'newValue');
```

Comment:
Could this be simplified to be?:
```js
this.active.parameters = {...this.active.parameters, ['newParameter' + this.active.parameters.size]: 'newValue' }
```

But I guess it would be too much effort to change this structure in the rest of the code and to implement the deletion, so I guess your version is better.

Reply:
I tried to do it like this, it breaks my tests as the returned object is not of type `Map` anymore